src/screens/Library/AllVocabs/AllVocabs.jsx

```diff
@@ -27,13 +28,14 @@ const AllVocabs = () => {
   const { packageId, groupId } = useParams();
 
   const [data, setData] = useState([]);
+  const [search, setSearch] = useState("");
   const [currentVocab, setCurrentVocab] = useState(null);
   const [showVocabModal, setShowVocabModal] = useState(false);
   const [showDeleteConfirmationModal, setShowDeleteConfirmationModal] =
     useState(false);
 
   const fetchVocabs = useCallback(() => {
-    getGroupVocabulary(groupId).then((response) => {
+    getGroupVocabulary(groupId, search).then((response) => {
```

Comment:
can you please use the `useDebounce` hook to not over query the server

Reply:
Done